backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -251,57 +251,61 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
-    // by the backend.
-    //
-    // All of these redirections can be expressions and will be given "value-mapped" versions of their inputs, which
-    // means that if any fully qualified filenames are generated they will be container paths. As mentioned above this
-    // doesn't matter for standard input since that's ephemeral to the container, but standard output and error paths
-    // will need to be mapped back to host paths for use outside the command script.
-    //
-    // Absolutize any redirect and overridden paths. All of these files must have absolute paths since the command script
-    // references them outside a (cd "execution dir"; ...) subshell. The default names are known to be relative paths,
-    // the names from redirections may or may not be relative.
-    def absolutizeContainerPath(path: String): String = {
-      if (path.startsWith(cwd.pathAsString)) path else cwd.resolve(path).pathAsString
-    }
-
-    val executionStdin = instantiatedCommand.evaluatedStdinRedirection map absolutizeContainerPath
-    val executionStdout = instantiatedCommand.evaluatedStdoutOverride.getOrElse(jobPaths.defaultStdoutFilename) |> absolutizeContainerPath
-    val executionStderr = instantiatedCommand.evaluatedStderrOverride.getOrElse(jobPaths.defaultStderrFilename) |> absolutizeContainerPath
-
-    def hostPathFromContainerPath(string: String): Path = {
-      val cwdString = cwd.pathAsString.ensureSlashed
-      val relativePath = string.stripPrefix(cwdString)
-      jobPaths.callExecutionRoot.resolve(relativePath)
-    }
 
+  /*
+   * Ensures the standard paths are correct w.r.t overridden paths. This is called in two places: when generating the command and
+   * before sending the results back to the engine. The latter is to cover for the case of a restart where the command is not re-generated,
+   * but the standard paths are still being used in the Success response to the engine. In that response paths need to be correct, hence
+   * the call to this method. The var is to avoid doing the work twice: in non restarted runs the command is generated and so there is no need
+   * to re-do this before sending the response.
+   */
+  private var jobPathsUpdated: Boolean = false
+  private def updateJobPaths() = if (!jobPathsUpdated) {
     // .get's are safe on stdout and stderr after falling back to default names above.
     jobPaths.standardPaths = StandardPaths(
       output = hostPathFromContainerPath(executionStdout),
       error = hostPathFromContainerPath(executionStderr)
     )
-
     // Re-publish stdout and stderr paths that were possibly just updated.
     tellMetadata(jobPaths.standardOutputAndErrorPaths)
+    jobPathsUpdated = true
+  }
+
+  def hostPathFromContainerPath(string: String): Path = {
+    val cwdString = cwd.pathAsString.ensureSlashed
+    val relativePath = string.stripPrefix(cwdString)
+    jobPaths.callExecutionRoot.resolve(relativePath)
+  }
+  // End added lift code
```

Comment:
cruft?

Reply:
It was already [there](https://github.com/broadinstitute/cromwell/pull/3827/files#diff-ce7445d19036fba5a706e2c697e937e3L261), not sure what it means so I left it. Might not be in the right place anymore though